ironfish/src/account/accounts.ts

```diff
@@ -517,14 +529,31 @@ export class Accounts {
             blockHash,
             initialNoteIndex: initialNoteIndex,
           },
-          outdatedAccounts,
+          accounts,
         )
         scan.onTransaction.emit(sequence)
       }
 
-      for (const account of outdatedAccounts) {
+      for (const account of accounts) {
         await this.updateHeadHash(account, blockHeader.hash)
       }
+
+      const hashHex = blockHeader.hash.toString('hex')
+
+      let i = 0
+      while (i < remainingAccounts.length) {
```

Comment:
nit: I'm not a huge fan of mutating objects (specifically deletion) while iterating over them using the index + length. Do you think this can be refactored to a `for` loop? I think it can become more readable